rocAL/rocAL_pybind/examples/image_comparison.py

```diff
@@ -0,0 +1,148 @@
+        total_valid_pixel_count = width * height * 1
+        for wt in range(width):
+            for ht in range(height):
+                ht = ht + image_offset
+                if pixels1[wt,ht] != pixels2[wt,ht]:
+                    pixel = abs(pixels1[wt,ht]-pixels2[wt,ht])
+                    if(pixel > 4):
+                        pixel = 5
+                    pixel_difference[pixel] += 1
+                else:
+                    pixel_difference[0] += 1
+    return pixel_difference, total_valid_pixel_count
+
+def main():
+    timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S_")
+    handlers = [logging.FileHandler('./rocal_unittest_log_file_' + timestamp + '.log'), logging.StreamHandler()]
+    logging.basicConfig(level = logging.INFO, handlers = handlers)
+    if( len(sys.argv) < 3 ):
+        logging.error('Pass atleast 2 arguments (python image_comparison.py <ref_output_folder_path> <rocal_ouput_folder_path>')
+        exit()
+
+    # Open the two images
+    ref_output_path = sys.argv[1]
+    rocal_output_path = sys.argv[2]
+
+    if not (os.path.exists(ref_output_path) and os.path.exists(rocal_output_path)):
+        logging.error("Path does not Exists")
+        exit()
+
+    total_case_count = 0
+    passed_case_count = 0
+    failed_case_count = 0
+    failed_case_list = []
+    golden_output_dir_list = os.listdir(ref_output_path)
+    rocal_output_dir_list = os.listdir(rocal_output_path)
+    randomized_augmentation = ["Snow", "Rain", "Jitter", "SNPNoise"]
+    golden_file_path = ""
+    for aug_name in rocal_output_dir_list:
+        temp = aug_name.split('.')
+        file_name_s = temp[0].split('_')
+        if(len(file_name_s) > 3):
+            file_name_s.pop()
+            golden_file_path = "_".join(file_name_s) + ".png"
+        else:
+            golden_file_path = aug_name
+
+        #For randomized augmentation
+        if(file_name_s[0] in randomized_augmentation):
+            total_case_count=total_case_count+1
+            augmentation_name = aug_name.split('.')[0]
+            logging.info("Running %s",augmentation_name)
+            passed_case_count=passed_case_count+1
+            logging.info("PASSED ")
+        elif golden_file_path in golden_output_dir_list:
+            total_case_count=total_case_count+1
+            ref_file_path = ref_output_path+golden_file_path
+            rocal_file_path = rocal_output_path+aug_name
+            if(os.path.exists(rocal_file_path) and os.path.exists(ref_file_path)):
+                logging.info("Running %s ",aug_name.split('.')[0])
+                img1 = Image.open(ref_file_path)
+                img2 = Image.open(rocal_file_path)
+
+                # Check if the images have the same dimensions
+                if img1.size != img2.size:
+                    logging.info("The images have different sizes.")
+                    exit()
+
+                # Compare the pixel values for each image
+                pixeldiff = None
```

Comment:
please rename this variable to pixel_diff
There are lot of styling issues i am seeing in this file. Please go through the entire tile and edit wherever applicable